ext/skeleton/skeleton.c

```diff
@@ -45,7 +45,7 @@ PHP_FUNCTION(confirm_extname_compiled)
 
 	len = spprintf(&strg, 0, "Congratulations! You have successfully modified ext/%.78s/config.m4. Module %.78s is now compiled into PHP.", "extname", arg);
 
-	RETVAL_STRINGL(strg, len);
+	RETVAL_STRING(strg, len);
 	efree(strg);
```

Comment:
@wangxx2026 It avoid reallocating the string. The code should look like this:

    RETURN_STR(strpprintf(0, "Congratulations! You have successfully modified ext/%.78s/config.m4. Module %.78s is now compiled into PHP.", "extname", arg));

Then you can drop the  `efree(strg)`, as well as the `strg` and `len` variables.